src/context/auth-context.jsx

```diff
@@ -0,0 +1,97 @@
+import { createContext, useReducer, useState } from "react";
+import axios from "axios";
+import { PassWordNotShowIcon } from "../icons/Icons";
+import { AuthReducer } from "../reducer/AuthReducer";
+const AuthContext = createContext();
+
+const AuthProvider = ({ children }) => {
+  const [userState, setUserState] = useState(false);
+
+  const [state, dispatch] = useReducer(AuthReducer, {
+    field: {},
+    passwordType: "password",
+    showPasswordIcon: <PassWordNotShowIcon />,
+    emailErrState: false,
+    passwordErrState: false,
+    confirmPasswordErrState: false,
+  });
+
+  const signupHandler = async (e) => {
+    e.preventDefault();
+    if (
+      state.field.email.indexOf("@") >= 0 &&
+      state.field.password.length >= 8 &&
+      state.field.confirmPassword === state.field.password
+    ) {
+      try {
+        const response = await axios.post(`/api/auth/signup`, state.field);
+        if (response.status === 201) {
+          setUserState(true);
+        }
+        console.log(response);
+        console.log(state.userState);
```

Comment:
Also, remove these `console.log` if not needed

Reply:
sure